libsolidity/formal/SMTEncoder.cpp

```diff
@@ -1297,8 +1297,7 @@ void SMTEncoder::indexOrMemberAssignment(Expression const& _expr, smtutil::Expre
 				if (var->hasReferenceOrMappingType())
 					resetReferences(*var);
 
-				m_context.addAssertion(m_context.newValue(*var) == _rightHandSide);
-				m_context.expression(_expr)->increaseIndex();
```

Comment:
I think this one should stay

Reply:
The same situation as the other two delete lines that are followed by `defineExpr`.